Sources/AuthFoundation/Migration/Migrators/OIDCLegacyMigrator.swift

```diff
@@ -0,0 +1,270 @@
+                                    scopes: String)
+        {
+            SDKVersion.register(migrator: LegacyOIDC(issuer: issuer,
+                                                     clientId: clientId,
+                                                     redirectUri: redirectUri,
+                                                     scopes: scopes))
+        }
+        
+        private static func register(_ config: OAuth2Client.PropertyListConfiguration) throws {
+            guard let redirectUri = config.redirectUri else {
+                throw OAuth2Client.PropertyListConfigurationError.missingConfigurationValues
+            }
+            
+            self.register(issuer: config.issuer,
+                          clientId: config.clientId,
+                          redirectUri: redirectUri,
+                          scopes: config.scopes)
+        }
+        
+        let issuer: URL
+        let clientId: String
+        let redirectUri: URL
+        let scopes: String
+        private(set) var migrationItems: [Keychain.Search.Result]?
+        
+        init(issuer: URL, clientId: String, redirectUri: URL, scopes: String) {
+            self.issuer = issuer
+            self.clientId = clientId
+            self.redirectUri = redirectUri
+            self.scopes = scopes
+        }
+        
+        public var needsMigration: Bool {
+            guard let regex = accountIdRegex,
+                  let items = try? Keychain
+                .Search(service: "")
+                .list()
+            else {
+                return false
+            }
+            
+            let results = items.filter({ searchResult in
+                guard searchResult.service == "",
+                      regex.matches(in: searchResult.account, range: NSRange(location: 0, length: searchResult.account.count)).count == 1
+                else {
+                    return false
+                }
+                return true
+            })
+                
+            return !results.isEmpty
+        }
+        
+        public func migrate() throws {
+            guard let regex = accountIdRegex else {
+                return
+            }
+
+            try Keychain
+                .Search(service: "")
+                .list()
+                .filter({ searchResult in
+                    regex.matches(in: searchResult.account,
+                                  range: NSRange(location: 0, length: searchResult.account.count)).count == 1
+                }).map({ searchResult in
+                    let item = try searchResult.get()
+                    return (searchResult, try decode(item.value))
+                }).forEach({ (item: Keychain.Search.Result, oldModel: StateManager?) in
+                    try importToken(item, from: oldModel)
+                })
+        }
+
+        private func decode(_ data: Data) throws -> StateManager? {
+            let archiver: NSKeyedUnarchiver
+            if #available(iOS 11.0, *) {
```

Comment:
maybe consider dropping support below 11?

Reply:
The SDK currently supports versions down to iOS 9, so backwards compatibility is still important.

Reply:
Wow. Does the new SDK still need to keep that level of support?

Reply:
FYI @stevelind-okta you might be interested to see our [OS version support policy](https://github.com/okta/okta-mobile-swift#platforms), the key feature of which is that we don't remove support for an OS version unless there's a compelling reason not to do so.  We only support down to iOS 11 officially, and dropping support for older versions isn't considered a breaking change.